src/main/java/TaskList.java

```diff
@@ -0,0 +1,167 @@
+     *
+     * @param list Arraylist that contains all the tasks.
+     */
+    public TaskList(ArrayList<Task> list) {
+        listOfTasks = list;
+    }
+
+    /**
+     * Deletes the specified tasks off the task list.
+     *
+     * @param text Delete command with the which tasks to delete based of their numbering.
+     * @throws IOException If the named file exists but is a directory rather than a regular file,
+     * does not exist but cannot be created, or cannot be opened for any other reason.
+     */
+    public void deleteCommand(String text) throws IOException {
+        int num = text.indexOf(" ");
+        int taskNumber = Integer.parseInt(text.substring(num + 1, num + 2));
+        tasking.printRemove();
+        Ui.printDelete(taskNumber);
+        listOfTasks.remove(taskNumber - 1);
+        store.writeToFile(Storage.file, "");
+        for (Task task : listOfTasks) {
+            store.addToFile(Storage.file, task.toString());
+        }
+        tasking.printNumOfTasks();
+    }
+
+    /**
+     * Deletes everything off the task list.
+     *
+     * @throws IOException If the named file exists
+     * but is a directory rather than a regular file,
+     * does not exist but cannot be created,
+     * or cannot be opened for any other reason.
+     */
+    public void deleteAllCommand(String text) throws IOException {
+        store.writeToFile(Storage.file, "");
+        listOfTasks.clear();
+        Ui.printLine();
+        Ui.printIndent();
+        System.out.println("Everything in your list has been removed! " +
+                "Add more tasks to get started again!!!");
+        Ui.printLine();
+    }
+
+    /**
+     * Adds todo task into arraylist and into file.
+     *
+     * @param text Takes in todo command with task.
+     * @throws IOException If the named file exists but is a directory rather than a regular file,
+     * does not exist but cannot be created, or cannot be opened for any other reason.
+     */
+    public void toDoCommand(String text) throws IOException {
+        tasking.printGI();
+        Ui.printIndent();
+        int num = text.indexOf(" ");
+        Task task = new Todo(text.substring(num + 1));
+        System.out.println("  " + task.toString());
+        listOfTasks.add(task);
+        store.addToFile(Storage.file, task.toString());
+        tasking.printNumOfTasks();
+    }
+
+    /**
+     * Event command with event and date at which it is and adds
+     * it into the arraylist and file.
+     *
+     * @param text Takes in a event command with the task and date.
+     * @throws DukeException If the user inputs more than the normal
+     * number of days/months/time, this error will be thrown out instead.
+     * @throws IOException If the named file exists but is a directory rather than a regular file,
+     * does not exist but cannot be created, or cannot be opened for any other reason.
+     */
+    public void eventCommand(String text) throws DukeException, IOException {
```

Comment:
I think the parsing of the String should be moved to the Parser class, the TaskList class should be concerned with the operations on the list itself.